src/component/feature/FeatureStrategy/FeatureStrategyEmpty/CopyButton/CopyButton.tsx

```diff
@@ -0,0 +1,107 @@
+import { MouseEvent, useContext, useState, VFC } from 'react';
+import {
+    Button,
+    ListItemIcon,
+    ListItemText,
+    Menu,
+    MenuItem,
+    Tooltip,
+} from '@mui/material';
+import { Lock } from '@mui/icons-material';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import { IFeatureEnvironment } from 'interfaces/featureToggle';
+import AccessContext from 'contexts/AccessContext';
+import { CREATE_FEATURE_STRATEGY } from 'component/providers/AccessProvider/permissions';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+interface ICopyButtonProps {
+    environmentId: IFeatureEnvironment['name'];
+    environments: IFeatureEnvironment['name'][];
+    onClick: (environmentId: string) => void;
+}
+
+export const CopyButton: VFC<ICopyButtonProps> = ({
+    environmentId,
+    environments,
+    onClick,
+}) => {
+    const projectId = useRequiredPathParam('projectId');
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const open = Boolean(anchorEl);
+    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+    const { hasAccess } = useContext(AccessContext);
+
+    const enabled = environments.some(environment =>
+        hasAccess(CREATE_FEATURE_STRATEGY, projectId, environment)
+    );
+
+    return (
+        <div>
+            <Tooltip title={enabled ? '' : '(Access denied)'}>
+                <div>
+                    <Button
```

Comment:
Could we use PermissionButton here?

Reply:
`PermissionButton` has built-in logic for access permission related to single item, not to an array of items.

Reply:
Yeah, you're right about that.